imcsdk/imccoremeta.py

```diff
@@ -48,6 +48,17 @@ def __init__(self, version):
         self.__patch = None
         self.__spin = None
 
+        # Matches major.minor(mr.patch), example 4.3(2.240002), new style of release numbering
+        match_pattern = re.compile("^(?P<major>[1-9][0-9]{0,2})\."
+                                   "(?P<minor>(([0-9])|([1-9][0-9]{0,1})))\("
+                                   "(?P<mr>(([0-9])|([1-9][0-9]{0,2})))\."
+                                   "(?P<patch>(([0-9]){6}))\)$")
```

Comment:
is this same as increasing one digit in the below expression

```
^(?P<major>[1-9][0-9]{0,2})\.(?P<minor>(([0-9])|([1-9][0-9]{0,1})))\((?P<mr>(([0-9])|([1-9][0-9]{0,2})))\.(?P<patch>(([0-9])|([1-9][0-9]{0,5})))\)$
```

Reply:
is 4.3(2.040002)  a legal patch version now?

Reply:
Just to be clear, these regex are matching CIMC firmware versions.

> is this same as increasing one digit for patch in the below expression

Yes good point.  It could be as simple as increasing the previous internal patch matching regex.  

> is 4.3(2.040002) a legal patch version now?

No not really, 04 would imply year of 2004 and while UCS was a product in 2004, it didn't use those patch numbers back then even for internal builds and if they did, it is unlikely the builds would have survived 10 years.  For the most part patches that are released used to have a single lower case alpha character ([a-z]).  It was only engineering builds of CIMC that would have used the numeric patch field and at one point those were single digits then changed to 5 digits.

The new release numbering for CIMC firmware is documented here:

https://www.cisco.com/c/en/us/td/docs/unified_computing/ucs/release/notes/b_release-notes-for-cisco-ucs-rack-server-software-release-4_3_2.html#cisco-imc-release-number-and-.iso-image-names

The new patch field is now 6 digits with a two digit year and a 4 digit build number.